### PR TITLE
DomainParticipant listener methods

### PR DESCRIPTION
**Pull request by laura-eprosima**
* set_listener and get_listener adapted to follow the DDS Standard
* New unit test to check PIM functionality
* PSM listener methods implementation
* Add unit test to check PSM functionality

NOTE: This PR ,ust be merged after #1073

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1621/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1387/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1617/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1629/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1395/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1625/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1644/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1410/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1640/)

File: test/unittest/dds/participant/ParticipantTests.cpp [hunk 1]
```diff
@@ -63,10 +63,10 @@ TEST(ParticipantTests, DomainParticipantPSMListenerMethods)
 
     ASSERT_EQ(participant.listener(), nullptr);
 
-    ::dds::domain::DomainParticipantListener* listener;
-    participant.listener(listener, ::dds::core::status::StatusMask::all());
+    ::dds::domain::DomainParticipantListener listener = ::dds::domain::DomainParticipantListener();
```
Comment (IkerLuengo) [inline test/unittest/dds/participant/ParticipantTests.cpp]:
```suggestion
    ::dds::domain::DomainParticipantListener listener;
```
No need to construct-and-assign.

File: test/unittest/dds/participant/ParticipantTests.cpp [hunk 2]
```diff
@@ -49,10 +49,10 @@ TEST(ParticipantTests, DomainParticipantListenerMethods)
 
     ASSERT_EQ(participant->get_listener(), nullptr);
 
-    DomainParticipantListener* listener;
-    participant->set_listener(listener);
+    DomainParticipantListener listener = DomainParticipantListener();
```
Comment (IkerLuengo) [inline test/unittest/dds/participant/ParticipantTests.cpp]:
```suggestion
    DomainParticipantListener listener;
```
No need to construct-and-assign.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1686/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1447/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1678/)

**Pull request closed without merging**